src/test/base_test.hpp

```diff
@@ -163,6 +163,29 @@ class BaseTestWithParam
     return chunk_encoding_spec;
   }
 
+  static void verify_encoding(const std::shared_ptr<Chunk>& chunk, const ChunkEncodingSpec& spec) {
```

Comment:
Maybe `bool chunk_matches_encoding`?

Reply:
The current function does several checks internally which would then directly lead to the faulted assertion. Returning a single bool would require us to have additional means (like table printing) to show the actual problem. 
Any reason not to go the direct path here?

Reply:
We can do that as well, but then I would make the method's name more speaking. Maybe `assert_chunk_encoding`?

Reply:
Sounds good. Will change.

Reply:
Done.